packages/rule-tester/src/utils/SourceCodeFixer.ts

```diff
@@ -63,7 +63,7 @@ export function applyFixes(
     const end = fix.range[1];
 
     // Remain it as a problem if it's overlapped or it's a negative range
-    if (lastPos >= start || start > end) {
+    if (lastPos > start || start > end) {
```

Comment:
The actual fix that causes all of this mess 😅

Reply:
https://github.com/eslint/eslint/blob/1f3744278433006042b8d5f4e9e1e488b2bbb011/lib/linter/source-code-fixer.js#L92

As per the comment at the top of the file - this code is directly forked from eslint core.

If it's broken here then this only changes tests run using our rule tester. Code run through eslint will have the old logic.

So either you've found a bug upstream or tests are running correctly, but correctly, but differently to how you expect.

If it's the former - the fix needs to land upstream before we land it here.